src/exts/tvdb_info/ui.py

```diff
@@ -0,0 +1,485 @@
+
+    def __init__(
+        self, bot: Bot, user_id: int, series: Series, watched_list: UserList, favorite_list: UserList
+    ) -> None:
+        super().__init__(timeout=None)
+        self.bot = bot
+        self.user_id = user_id
+        self.series = series
+        self.user: User
+        self.watched_list = watched_list
+        self.favorite_list = favorite_list
+        self.episodes: dict[int, list[Episode]]
+        self.season_idx: int = 1
+        self.episode_idx: int = 1
+
+        self._reactive_buttons_store: list[ReactiveButton] | None = None
+
+        self.episode_dropdown = discord.ui.Select(
+            placeholder="Select an episode",
+        )
+        self.episode_dropdown.callback = self._episode_dropdown_callback
+        self.add_item(self.episode_dropdown)
+
+        self.season_dropdown = discord.ui.Select(
+            placeholder="Select a season",
+        )
+        self.season_dropdown.callback = self._season_dropdown_callback
+        self.add_item(self.season_dropdown)
+
+        self.watched_btn = discord.ui.Button(
+            row=2,
+        )
+        self.watched_btn.callback = self._watched_callback
+        self.add_item(self.watched_btn)
+
+        self.favorite_btn = discord.ui.Button(
+            row=2,
+        )
+        self.favorite_btn.callback = self._favorite_callback
+        self.add_item(self.favorite_btn)
+
+    @property
+    @override
+    def _reactive_buttons(self) -> list[ReactiveButton]:
+        if self._reactive_buttons_store:
+            return self._reactive_buttons_store
+        return [
+            (
+                self.watched_btn,
+                self._current_watched_list_item,
+                {
+                    "label": "Mark as watched",
+                    "style": discord.ButtonStyle.success,
+                    "emoji": "✅",
+                },
+                {
+                    "label": "Unmark as watched",
+                    "style": discord.ButtonStyle.primary,
+                    "emoji": "❌",
+                },
+            ),
+            (
+                self.favorite_btn,
+                self._current_favorite_list_item,
+                {
+                    "label": "Favorite",
+                    "style": discord.ButtonStyle.primary,
+                    "emoji": "⭐",
+                },
+                {
+                    "label": "Unfavorite",
+                    "style": discord.ButtonStyle.secondary,
+                    "emoji": "❌",
+                },
+            ),
+        ]
+
+    @property
+    def _current_episode(self) -> "Episode":
+        return self.episodes[self.season_idx][self.episode_idx - 1]
+
+    @property
+    def _current_watched_list_item(self) -> UserListItem | None:
+        return get_first(
+            item
+            for item in self.watched_list.items
+            if item.tvdb_id == self._current_episode.id and item.kind == UserListItemKind.EPISODE
+        )
```

Comment:
Hmm, a direct lookup would be better than going over everything in the watched list. Some people could have over 10k episodes watched, and this list will contain all of that, looping over each item and comparing could actually cause slow-downs here.

Same remark for favorite list, although the amount of favoritted items will very likely be much smaller, so it's not as important there, but still.